discord-scripts/linear-integration.ts

```diff
@@ -0,0 +1,378 @@
+          type: ApplicationCommandOptionType.Subcommand,
+          description: "Connects a Linear team to this channel.",
+          options: [
+            {
+              name: "team",
+              type: ApplicationCommandOptionType.String,
+              description: "The ID of the Linear team to connect.",
+              required: true,
+              choices: teams.map((team) => ({
+                name: team.name,
+                value: team.id,
+              })),
+            },
+          ],
+        },
+        {
+          name: DISCONNECT_SUBCOMMAND_NAME,
+          type: ApplicationCommandOptionType.Subcommand,
+          description: "Disconnects a Linear team from this channel.",
+          options: [
+            {
+              name: "team",
+              type: ApplicationCommandOptionType.String,
+              description: "The ID of the Linear team to disconnect.",
+              required: true,
+              choices: teams.map((team) => ({
+                name: team.name,
+                value: team.id,
+              })),
+            },
+          ],
+        },
+      ],
+    })
+    robot.logger.info("Created linear command.")
+  }
+
+  discordClient.on("interactionCreate", async (interaction) => {
+    robot.logger.info("Received interaction:", interaction)
+    if (
+      interaction.isChatInputCommand() &&
+      interaction.commandName === COMMAND_NAME
+    ) {
+      robot.logger.info("Running command")
+      const subcommand = interaction.options.getSubcommand()
+      const teamId = interaction.options.getString("team")
+
+      if (!teamId) {
+        await interaction.reply({
+          content: "Please provide a valid team ID.",
+          ephemeral: true,
+        })
+        return
+      }
+
+      const existingConnections =
+        robot.brain.get(LINEAR_BRAIN_KEY)?.connections ?? {}
+
+      if (subcommand === CONNECT_SUBCOMMAND_NAME) {
+        const { channel } = interaction
+        if (!(channel instanceof TextChannel)) {
+          await interaction.reply({
+            content: "This command must be used in a text channel.",
+            ephemeral: true,
+          })
+          return
+        }
+        if (existingConnections[teamId]) {
+          await interaction.reply({
+            content: "⚠️ This Linear team is already connected to a channel.",
+            ephemeral: true,
+          })
+          return
+        }
+        await interaction.reply({
+          content: "Linked this channel",
+          ephemeral: true,
+        })
+
+        robot.logger.info("Creating custom webhook URL for team:", teamId)
+        const customWebhookUrl = await createCustomWebhookUrl(channel)
+
+        if (customWebhookUrl) {
+          const team = await linearClient.team(teamId)
+          const sanitizedChannelName = channel.name
+            .toLowerCase()
+            .replace(/\s+/g, "-")
+          const sanitizedTeamName = team.name.toLowerCase().replace(/\s+/g, "-")
```

Comment:
We're doing this kind of thing in a couple of places, wonder if we should pull it to a helper if we don't already have one.